### DISCUSSION TITLE
How to access the search filtered results?

### DISCUSSION THREAD

**Discussion by hermbo** [JedWatson/react-select#4767] [Q&A] [answered]
Hello I am trying to gain access to the filtered results when a user starts typing into the search bar.

Example, if I start typing `xar` into the search bar and then it filters back 2 results out of a total of 50 how do I access those 2 results?

Screenshot:
<img width="329" alt="Screen Shot 2021-09-10 at 1 33 15 PM" src="https://user-images.githubusercontent.com/38641518/132914470-6ce9b718-0c4b-4fa8-8fd2-dd13c4f0f0db.png">

**Comment by Rall3n** [comment, marked as answer]
Hello @hermbo,

the `Select` component has a method called `getFocusableOptions` which returns all options which are currently visible inside the list.
**Be aware** that the function only returns the options if the menu is open.

```jsx
const MySelect = () => {
  const selectRef = useRef(null);
  
  // Use selectRef.select.getFocusableOptions() to access options.

  return <Select ref={selectRef} />;
}
```

**Reply by hermbo** [reply]
Interesting, I didn't seem to find that in the documentation, how did you come across it?

**Reply by Rall3n** [reply]
By looking through the source code (and by having knowledge of it).

It is not exactly intended for public usage, but it is still usable.
